platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -238,7 +249,7 @@ protected void writeLongsData(final FileChannel fc) throws IOException {
         final ByteBuffer transferBuffer = initOrGetTransferBuffer();
         final int totalNumOfChunks = calculateNumberOfChunks(size());
         final long currentMinValidIndex = minValidIndex.get();
-        final int firstChunkWithDataIndex = (int) currentMinValidIndex / numLongsPerChunk;
+        final int firstChunkWithDataIndex = (int) (currentMinValidIndex / numLongsPerChunk);
```

Comment:
Here was the overflow bug

Reply:
Probably should use `Math.toIntExact()` instead of just a cast if we need to be sure it does not overflow (that method throws if it would overflow).

Reply:
Fair enough. I'll update the code